sklearn/decomposition/incremental_pca.py

```diff
@@ -190,7 +195,8 @@ def partial_fit(self, X, y=None):
         self: object
             Returns the instance itself.
         """
-        X = check_array(X, copy=self.copy, dtype=np.float)
+        if check_input:
+            X = check_array(X, copy=self.copy, dtype=[np.float64, np.float32])
```

Comment:
Not to be fixed here but it seems that X is modified inplace in `transform` making the copy attribute meaningless for transform.